packages/firestore/src/api/cache_config.ts

```diff
@@ -23,86 +23,185 @@ import {
   OnlineComponentProvider
 } from '../core/component_provider';
 
-export interface MemoryLocalCache {
+/* eslint @typescript-eslint/consistent-type-definitions: ["error", "type"] */
+/**
+ * Provides a in-memory cache to the SDK. This is the default cache unless explicitly
+ * specified otherwise.
+ *
+ * To use, create an instance using the factory function `memoryLocalCache()`, then
+ * set the instance to `FirestoreSettings.cache` and call `initializeFirestore` using
+ * the settings object.
+ */
+export type MemoryLocalCache = {
   kind: 'memory';
+  /**
+   * @internal
+   */
   _onlineComponentProvider: OnlineComponentProvider;
+  /**
+   * @internal
+   */
   _offlineComponentProvider: MemoryOfflineComponentProvider;
-}
+};
 
 class MemoryLocalCacheImpl implements MemoryLocalCache {
   kind: 'memory' = 'memory';
+  /**
+   * @internal
+   */
   _onlineComponentProvider: OnlineComponentProvider;
+  /**
+   * @internal
+   */
   _offlineComponentProvider: MemoryOfflineComponentProvider;
 
   constructor() {
     this._onlineComponentProvider = new OnlineComponentProvider();
     this._offlineComponentProvider = new MemoryOfflineComponentProvider();
   }
+
+  toJSON(): {} {
+    return { kind: this.kind };
+  }
 }
 
-export interface IndexedDbLocalCache {
+/**
+ * Provides a cache backed by IndexedDb to the SDK.
+ *
+ * To use, create an instance using the factory function `indexedDbLocalCache()`, then
```

Comment:
Consider using `{@link indexDbLocalCache()}` to generate a link in documentation to this factory. Also can apply above to memory cache.